app/internal/page/sessions.go

```diff
@@ -0,0 +1,170 @@
+package page
+
+import (
+	"encoding/gob"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/sessions"
+)
+
+type MissingSessionError string
+
+func (e MissingSessionError) Error() string {
+	return fmt.Sprintf("missing %s session", string(e))
+}
+
+type InvalidSessionError string
+
+func (e InvalidSessionError) Error() string {
+	return fmt.Sprintf("%s session invalid", string(e))
+}
+
+var (
+	sessionCookieOptions = &sessions.Options{
+		Path:     "/",
+		MaxAge:   24 * 60 * 60,
+		SameSite: http.SameSiteLaxMode,
+		HttpOnly: true,
+		Secure:   true, // check this works
```

Comment:
Did it work? 🤔

Reply:
ha, yes, and this is why I should put TODO before my comments so I actually spot them